File: Android/Sources/UserControlLibraries/GaodeMaps/src/main/java/com/genexus/android/controls/maps/gaode/GxMapViewData.java
package com.genexus.android.controls.maps.gaode;

import com.genexus.android.core.base.model.Entity;
import com.genexus.android.core.controllers.ViewData;
import com.genexus.android.core.controls.maps.GxMapViewDefinition;
import com.genexus.android.core.controls.maps.common.MapDataBase;
import com.genexus.android.core.controls.maps.common.MapLayer;

import java.util.List;

class GxMapViewData extends MapDataBase<GxMapViewItem, MapLocation, MapLocationBounds> {
	private static final long serialVersionUID = 1L;

	public GxMapViewData(GxMapViewDefinition mapDefinition, ViewData itemsData) {
		super(itemsData, new MapUtils(mapDefinition));
	}

	@Override
	protected GxMapViewItem newMapItem(List<MapLocation> locationList, Entity itemData, MapLayer.FeatureType featureType,
	                                   String layerId, String themeClassName) {
		return new GxMapViewItem(locationList, itemData, featureType, layerId, themeClassName);
	}
}

File: Android/Sources/UserControlLibraries/MapboxLib/src/main/java/com/genexus/android/controls/maps/mapbox/GxMapViewData.kt
package com.genexus.android.controls.maps.mapbox

import com.genexus.android.core.base.model.Entity
import com.genexus.android.core.controllers.ViewData
import com.genexus.android.core.controls.maps.GxMapViewDefinition
import com.genexus.android.core.controls.maps.common.MapDataBase
import com.genexus.android.core.controls.maps.common.MapLayer.FeatureType

internal class GxMapViewData(mapDefinition: GxMapViewDefinition?, itemsData: ViewData?) :
    MapDataBase<GxMapViewItem, MapLocation, MapLocationBounds>(itemsData, MapUtils(mapDefinition)) {

    override fun newMapItem(
        locationList: MutableList<MapLocation?>,
        itemData: Entity,
        featureType: FeatureType,
        layerId: String?,
        themeClassName: String?
    ): GxMapViewItem {
        return GxMapViewItem(locationList, itemData, featureType, layerId, themeClassName)
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
